v2/cmd/samples/http/receiver-protected/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudevents/sdk-go/v2/protocol/http"
+	"log"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+)
+
+func main() {
+	ctx := context.Background()
+	p, err := cloudevents.NewHTTP()
+	if err != nil {
+		log.Fatalf("failed to create protocol: %s", err.Error())
+	}
+
+	rate := 100
+	p.WebhookConfig = &http.WebhookConfig{
+		AllowedMethods:  []string{"POST", "OPTIONS"},
+		AllowedRate:     &rate,
+		AutoACKCallback: true,
+		AllowedOrigins:  []string{"http://localhost:8181"},
+	}
+	p.OptionsHandlerFn = p.OptionsHandler
```

Comment:
?

Reply:
Can this be a default?

Reply:
yeah a future PR can add this to the default client to respond to OPTIONS

Reply:
I think it should be opt-in, because you need to know the rate limit and such

Reply:
plus the default allowed origin would have to be "*" which seems like a bad default

Reply:
Do you think it should be an opt-in or an opt-out feature? I don't think this interfere with the average usage of sdk-go

Reply:
because OPTIONS is optional by the spec, it should be OPT-IN so that integrators have to make choices about rate limits and origin filters up front.

Reply:
Okok i feel we should document it and make a "better entrypoint", like  and option for the protocol `EnableWebHookSupport()` or something like that that does this assignment